### PR TITLE
[Serving] Fix TIR RoPE issue

### PR DESCRIPTION
**Pull request by MasterJH5574**
PR #1570 was merged before testing. Some issues are caught during testing and this PR fixes the issues.

Validated that with this fix, the TIR kernels pass end-to-end tests.

### CODE REVIEW

File: python/mlc_chat/nn/kv_cache.py [hunk 1]
```diff
@@ -329,23 +329,24 @@ def tir_rotary(
...
-                    for d0 in T.thread_binding(T.ceildiv(head_dim, 4), thread="threadIdx.x"):
-                        for d1 in T.vectorized(4):
-                            s: T.int32 = s0 * 32 + s1
-                            d: T.int32 = d0 * 4 + d1
-                            if s < append_len and d < head_dim:
-                                if h < num_q_heads:
-                                    q[s + instance_offset, h, d] = _rope(q, s, h, d, rope_offset, instance_offset)
-                                else:
-                                    k[s + instance_offset, h - num_q_heads, d] = _rope(k, s, h - num_q_heads, d, rope_offset, instance_offset)
+        with T.block():
```
Comment (MasterJH5574) [inline python/mlc_chat/nn/kv_cache.py:332]: The parser does not automatically add a block without `with T.block():`. Have not got a chance to see why.

File: python/mlc_chat/op/position_embedding.py [hunk 2]
```diff
@@ -141,16 +141,16 @@ def fused_rope(  # pylint: disable=too-many-locals
             with T.block("llama_fused_rope"):
                 b, s, h, d = T.axis.remap("SSSS", iters)
                 if h < num_q_heads:
-                    q[b, s, h, d] = (
-                        _rope(qkv, b, s, h, d, total_seq_len - seq_len)
-                        if d < rotary_dim
-                        else qkv[b, s, h, d]
```
Comment (MasterJH5574) [inline python/mlc_chat/op/position_embedding.py:147]: `B if A else C` does not naturally work with TIR parser.

File: python/mlc_chat/nn/kv_cache.py (hunk 1 above)
Comment (junrushao) [reply python/mlc_chat/nn/kv_cache.py:332]: BTW, why do we need a block here anyways given we've done scheduling already by hand?

File: python/mlc_chat/nn/kv_cache.py [hunk 3]
```diff
@@ -380,7 +381,7 @@ def _attention_prefill(h_kv, h_q, d, dtype):
     assert dtype == "float16", f"TIR attention kernel does not support dtype {dtype} right now"
     # pylint: disable=invalid-name
     NUM_BLKS = 16
-    LOAD_VEC = 8 // (DataType(dtype).bits + 7) / 8  # 8 bytes
+    LOAD_VEC = 8 // ((DataType(dtype).bits + 7) // 8)  # 8 bytes
```
Comment (junrushao) [inline python/mlc_chat/nn/kv_cache.py:384]: Thanks for catching this typo!

File: python/mlc_chat/op/position_embedding.py (hunk 2 above)
Comment (junrushao) [reply python/mlc_chat/op/position_embedding.py:147]: Seems something we should fix in the parser :))

File: python/mlc_chat/nn/kv_cache.py (hunk 1 above)
Comment (MasterJH5574) [reply python/mlc_chat/nn/kv_cache.py:332]: There is a pass which assuming all PrimFunc at that phase is not lowered https://github.com/mlc-ai/mlc-llm/blob/main/python/mlc_chat/compiler_pass/lift_global_buffer_alloc.py#L92.

**Review by junrushao — APPROVED** [review]

File: python/mlc_chat/nn/kv_cache.py (hunk 1 above)
Comment (junrushao) [reply python/mlc_chat/nn/kv_cache.py:332]: Thanks for the clarification!

**Pull request merged**